creation/views.py

```diff
@@ -2861,3 +2943,567 @@ def update_codefiles(request):
+
+        try :
+            contributor_create = ContributorRole()            
+            contributor_create.foss_category_id = tut.foss_id
+            contributor_create.language_id = lid 
+            contributor_create.status = 1
+            contributor_create.user_id = user.id
+            contributor_create.tutorial_detail_id = tut.id
+            contributor_create.save()
+        except:
+            contributor_update = ContributorRole.objects.filter(language_id = lid ,tutorial_detail_id = tut.id , user_id = user.id).update(
+                status=1)
+        try:
+            TutorialsAvailable.objects.filter(tutorial_detail = tut ,language = lid).delete()
+        except:
+            messages.error(request, "Tutorial not found in TutorialsAvailable, should be already present in TutorialResource")
+
+@login_required
+@csrf_exempt
+def allocate(request,tdid,lid,uid,days):
+
+    user = User.objects.get(id=uid)
+    uid= user.id
+    tut = TutorialDetail.objects.get(id=tdid)
+    data = "Response"
+    level_name ={
+    1:'Basic',
+    2:'Intermediate',
+    3:'Advanced' 
+    }
+    
+    final_query = TutorialsAvailable.objects.get(tutorial_detail_id = tut.id ,language = lid)
+    tuto = TutorialDetail.objects.filter(foss_id = final_query.tutorial_detail.foss_id,level_id = (final_query.tutorial_detail.level_id -1))
+    contrib = ContributorRating.objects.filter(user_id=uid,language_id=lid).values('rating')
+    check = check_contributor_eligibility(request, contrib)
+    contrib_tutorial_count = ContributorRole.objects.filter(user_id=uid,status=1).aggregate(Count('id'))
+    bid_count = TutorialResource.objects.filter(Q(script_user_id = uid)|Q(video_user_id=uid),language_id = lid ,assignment_status=1
+                ).exclude(language_id=22).aggregate(Count('id'))
+    if no_of_foss_gt_3(uid):
+        messages.error(request,"Maximum of 3 FOSSes allowed per user")
+        return HttpResponse(json.dumps(data), content_type='application/json')
+    elif check:        
+        if contrib[0]['rating']<3:
+            if contrib_tutorial_count['id__count'] > 3 :
+                if is_language_manager(request.user):
+                    messages.error(request, "You cannot allocate more than 3 tutorials to a contributor of rating less than 3")
+                else:
+                    messages.error(request,"You cannot allocate more than 3 tutorials ")
+            else:
+                single_tutorial_allocater(request,tuto,tut,lid,days,user)
+            
+            return HttpResponse(json.dumps(data), content_type='application/json')
+            
+        else:
+            single_tutorial_allocater(request,tuto,tut,lid,days,user)
+    
+    return HttpResponse(json.dumps(data), content_type='application/json')
+    #return HttpResponseRedirect(request.META['HTTP_REFERER'])
+    
+@csrf_exempt
+def extend_submission_date(request):
+    tutorial = request.POST.get('tutorial')
```

Comment:
If request is GET then this method will fail!

Reply:
Yes code updated with try catch